src/flow/src/adapter/table_source.rs

```diff
@@ -25,18 +28,58 @@ use crate::adapter::TableName;
 use crate::error::{
     Error, ExternalSnafu, TableNotFoundMetaSnafu, TableNotFoundSnafu, UnexpectedSnafu,
 };
-use crate::repr::RelationDesc;
+use crate::repr::{ColumnType, RelationDesc, RelationType};
+
+/// Table source but for flow, provide table schema by table name/id
+#[async_trait::async_trait]
+pub trait FlowTableSource: Send + Sync + std::fmt::Debug {
+    async fn table_name_from_id(&self, table_id: &TableId) -> Result<TableName, Error>;
+    async fn table_id_from_name(&self, name: &TableName) -> Result<TableId, Error>;
+
+    /// Get the table schema by table name
+    async fn table(&self, name: &TableName) -> Result<RelationDesc, Error> {
+        let id = self.table_id_from_name(name).await?;
+        self.table_from_id(&id).await
+    }
+    async fn table_from_id(&self, table_id: &TableId) -> Result<RelationDesc, Error>;
+}
 
 /// mapping of table name <-> table id should be query from tableinfo manager
-pub struct TableSource {
+#[derive(Clone)]
+pub struct KvBackendTableSource {
```

Comment:
I think it's fine to name it `TableSource` since the KvBackend is part of XXXManager.